packages/persist/lib/server.integration.test.ts

```diff
@@ -0,0 +1,207 @@
+import { expect, describe, it, beforeAll, afterAll } from 'vitest';
+import { server } from './server.js';
+import fetch from 'node-fetch';
+import {
+    multipleMigrations,
+    createActivityLog,
+    environmentService,
+    connectionService,
+    createSync,
+    createSyncJob,
+    AuthCredentials,
+    Connection,
+    Sync,
+    Job as SyncJob,
+    SyncType,
+    SyncStatus,
+    Environment,
+    db
+} from '@nangohq/shared';
+
+describe('Persist API', () => {
+    const port = 3096;
+    const serverUrl = `http://localhost:${port}`;
+    let seed: {
+        env: Environment;
+        activityLogId: number;
+        connection: Connection;
+        sync: Sync;
+        syncJob: SyncJob;
+    };
+
+    beforeAll(async () => {
+        await multipleMigrations();
+        seed = await initDb();
+        server.listen(port);
+    });
+
+    afterAll(async () => {
+        await clearDb();
+    });
+
+    it('should server /health', async () => {
+        const response = await fetch(`${serverUrl}/health`);
+        expect(response.status).toEqual(200);
+        expect(await response.json()).toEqual({ status: 'ok' });
+    });
+
+    it('should log', async () => {
+        const response = await fetch(`${serverUrl}/environment/${seed.env.id}/log`, {
+            method: 'POST',
+            body: JSON.stringify({ activityLogId: seed.activityLogId, level: 'info', msg: 'Hello, world!' }),
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        });
+        expect(response.status).toEqual(201);
+    });
+
+    describe('save records', () => {
+        it('should error if no records', async () => {
+            const response = await fetch(
+                `${serverUrl}/environment/${seed.env.id}/connection/${seed.connection.id}/sync/${seed.sync.id}/job/${seed.syncJob.id}/records`,
+                {
+                    method: 'POST',
+                    body: JSON.stringify({
+                        model: 'MyModel',
+                        records: [],
+                        providerConfigKey: seed.connection.provider_config_key,
+                        connectionId: seed.connection.connection_id,
+                        lastSyncDate: new Date(),
+                        trackDeletes: false,
+                        softDelete: true
+                    }),
+                    headers: {
+                        'Content-Type': 'application/json'
+                    }
+                }
+            );
+            expect(response.status).toEqual(400);
+            const respBody = (await response.json()) as any[];
+            expect(respBody[0]['errors']['issues'][0]['path'][0]).toEqual('records');
```

Comment:
not related to this pr: Could be interesting to test the full body with toMatchSnapshot or toMatchObject since we don't have a lot of tests regarding this